javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/model/typesystem/ReferenceTypeTest.java

```diff
@@ -641,36 +658,57 @@ void testDirectAncestorsOfInterfaceExtendingInterface() {
 
     @Test
     void testDirectAncestorsOfClassWithoutSuperClassOrInterfaces() {
-        ResolvedReferenceType buffer = new ReferenceTypeImpl(
-                new ReflectionClassDeclaration(Buffer.class, typeSolver), typeSolver);
-        Set<String> ancestors = buffer.getDirectAncestors().stream().map(a -> a.describe()).collect(Collectors.toSet());
-        assertEquals(new HashSet<>(Arrays.asList("java.lang.Object")), ancestors);
+        ResolvedReferenceType buffer = new ReferenceTypeImpl(new ReflectionClassDeclaration(Buffer.class, typeSolver), typeSolver);
+        Set<String> ancestors = buffer.getDirectAncestors()
+                .stream()
+                .map(ResolvedReferenceType::describe)
+                .collect(Collectors.toSet());
+
+        assertThat(ancestors, equalTo(new HashSet<>(Arrays.asList("java.lang.Object"))));
     }
 
     @Test
     void testDirectAncestorsOfObjectClass() {
-        ResolvedReferenceType object = new ReferenceTypeImpl(
-                new ReflectionClassDeclaration(Object.class, typeSolver), typeSolver);
-        Set<String> ancestors = object.getDirectAncestors().stream().map(a -> a.describe()).collect(Collectors.toSet());
+        ResolvedReferenceType object = new ReferenceTypeImpl(new ReflectionClassDeclaration(Object.class, typeSolver), typeSolver);
+        Set<String> ancestors = object.getDirectAncestors()
+                .stream()
+                .map(ResolvedReferenceType::describe)
+                .collect(Collectors.toSet());
+
         assertEquals(new HashSet<>(), ancestors);
     }
 
     @Test
     void testDirectAncestorsOfClassWithSuperClass() {
-        ResolvedReferenceType charbuffer = new ReferenceTypeImpl(
-                new ReflectionClassDeclaration(CharBuffer.class, typeSolver), typeSolver);
-        Set<String> ancestors = charbuffer.getDirectAncestors().stream().map(a -> a.describe()).collect(Collectors.toSet());
-        assertEquals(new HashSet<>(Arrays.asList("java.lang.CharSequence", "java.lang.Appendable",
-                "java.nio.Buffer", "java.lang.Readable", "java.lang.Comparable<java.nio.CharBuffer>")), ancestors);
+        ResolvedReferenceType charbuffer = new ReferenceTypeImpl(new ReflectionClassDeclaration(CharBuffer.class, typeSolver), typeSolver);
+        Set<String> ancestors = charbuffer.getDirectAncestors()
+                .stream()
+                .map(ResolvedReferenceType::describe)
+                .collect(Collectors.toSet());
+
+        assertThat(ancestors, containsInAnyOrder(
+                "java.lang.Object",
```

Comment:
FIXME: hmm... Did I muddle things up here by including `java.lang.Object` as an ancestor...?

Reply:
This is still unclear to me.

Other methods such as `getAllMethods()` make use of `getDirectAncestors()` where I would expect methods from implicitly extended classes (i.e. `java.lang.Object`) to be included within the output.

This assumption is contradicted by the javadoc for `getDirectAncestors()`, however, which explicitly states that implicit extends is excluded.

Bottom line: additional testing is required to determine the current behaviour versus this PR's behaviour versus the "desired"/"proper" behaviour (hopefully they are all the same!).

```java
    /**
     * Return direct ancestors, that means the superclasses and interfaces implemented directly.
     * This list should include Object if the class has no other superclass or the interface is not extending another interface.
     * There is an exception for Object itself.
     */
    public abstract List<ResolvedReferenceType> getDirectAncestors();
```

Reply:
Confirmed as a change in behaviour that I caused. 

I've just added the following comments to explain, noting that `getAncestors()` only returns direct ancestors (i.e. ignoring the implicit relationships such as implicitly extending java.lang.Object or java.lang.Enum).

```java 

        List<ResolvedReferenceType> ancestors = typeDeclaration.getAncestors();

        ancestors = ancestors.stream()
                .map(a -> typeParametersMap().replaceAll(a).asReferenceType())
                .collect(Collectors.toList());

        // Avoid repetitions of Object -- remove them all and, if we removed any, add it back precisely once.
        if(ancestors.removeIf(ResolvedReferenceType::isJavaLangObject)) {
            // We only want to add it back in if we have explicitly extended Object (ignoring implicit supertypes),
            //   thus only do this if it has been removed at least once from the ancestors.
            ResolvedReferenceTypeDeclaration objectType = typeSolver.getSolvedJavaLangObject();
            ResolvedReferenceType objectRef = create(objectType);
            ancestors.add(objectRef);
        }
```

I found this distinction confusing (i.e. the difference between `getAncestors` vs `getAllAncestors` vs `getDirectAncestors` vs what I presume should be named as `getExplicitAncestors` and `getImplicitAncestors`). 

#2696 has been opened to investigate this further.

Reply:
I've just edited this PR to undo this inadvertent change of behaviour, so this comment chain can be marked as "resolved"

Reply:
I have added a fix to undo this change of behaviour.

See also #2696 and https://github.com/javaparser/javaparser/pull/2693/files/5f06176f074805fcc32a0f18989b031af5aadae0#r429632045